src/test/java/org/prebid/server/ApplicationTest.java

```diff
@@ -246,193 +489,543 @@ public void openrtb2AuctionShouldRespondWithBidsFromDifferentExchanges() throws
+        // given
+        // rubicon bid response for imp 1
+        wireMockRule.stubFor(post(urlPathEqualTo("/rubicon-exchange"))
+                .withQueryParam("tk_xint", equalTo("rp-pbs"))
+                .withBasicAuth("rubicon_user", "rubicon_password")
+                .withHeader("Content-Type", equalToIgnoreCase("application/json;charset=utf-8"))
+                .withHeader("Accept", equalTo("application/json"))
+                .withHeader("User-Agent", equalTo("prebid-server/1.0"))
+                .withRequestBody(equalToJson(jsonFrom("openrtb2/rubicon&appnexus/test-rubicon-bid-request-1.json")))
+                .willReturn(aResponse().withBody(jsonFrom("openrtb2/rubicon&appnexus/test-rubicon-bid-response-1.json"))));
+
+        // rubicon bid response for imp 2
+        wireMockRule.stubFor(post(urlPathEqualTo("/rubicon-exchange"))
+                .withRequestBody(equalToJson(jsonFrom("openrtb2/rubicon&appnexus/test-rubicon-bid-request-2.json")))
+                .willReturn(aResponse().withBody(jsonFrom("openrtb2/rubicon&appnexus/test-rubicon-bid-response-2.json"))));
+
+        // appnexus bid response for imp 3
+        wireMockRule.stubFor(post(urlPathEqualTo("/appnexus-exchange"))
+                .withRequestBody(equalToJson(jsonFrom("openrtb2/rubicon&appnexus/test-appnexus-bid-request-1.json")))
+                .willReturn(aResponse().withBody(jsonFrom("openrtb2/rubicon&appnexus/test-appnexus-bid-response-1.json"))));
+
+        // appnexus bid response for imp 3 with alias parameters
+        wireMockRule.stubFor(post(urlPathEqualTo("/appnexus-exchange"))
+                .withRequestBody(equalToJson(jsonFrom("openrtb2/rubicon&appnexus/test-appnexus-bid-request-2.json")))
+                .willReturn(aResponse().withBody(jsonFrom("openrtb2/rubicon&appnexus/test-appnexus-bid-response-2.json"))));
+
+        // pre-bid cache
+        wireMockRule.stubFor(post(urlPathEqualTo("/cache"))
+                .withRequestBody(equalToJson(jsonFrom("openrtb2/rubicon&appnexus/test-cache-rubicon-appnexus-request.json")))
+                .willReturn(aResponse().withBody(jsonFrom("openrtb2/rubicon&appnexus/test-cache-rubicon-appnexus-response.json"))));
+
+        final Map<String, String> exchanges = new HashMap<>();
+        exchanges.put(RUBICON, "http://localhost:" + WIREMOCK_PORT + "/rubicon-exchange?tk_xint=rp-pbs");
+        exchanges.put(APPNEXUS, "http://localhost:" + WIREMOCK_PORT + "/appnexus-exchange");
+        exchanges.put(APPNEXUS_ALIAS, null);
+
+        // when
+        final Response response = given(spec)
+                .header("Referer", "http://www.example.com")
+                .header("X-Forwarded-For", "192.168.244.1")
+                .header("User-Agent", "userAgent")
+                .header("Origin", "http://www.example.com")
+                // this uids cookie value stands for
+//                {"uids":{"rubicon":"J5VLCWQP-26-CWFT","adnxs":"12345","audienceNetwork":"FB-UID","pulsepoint":"PP-UID",
```

Comment:
Only uids from `rubicon` and `appnexus` required here. If others were removed from actual cookie already - please update the comment